examples/Makefile

```diff
@@ -1,25 +1,13 @@
+## this is a simple wrapper just to forward on known commands to the
+## embedding example with values pulled from Make.inc
 SRCDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
-BUILDDIR := .
 JULIAHOME := $(abspath $(SRCDIR)/..)
+BUILDDIR := .
 include $(JULIAHOME)/Make.inc
 
-outdir := $(libexecdir)
-
-embedding_binary := $(abspath $(outdir)/embedding$(JULIA_LIBSUFFIX)$(EXE))
-
-release: embedding
-debug: embedding-debug
-
-embedding: $(embedding_binary)
-embedding-debug: $(embedding_binary)
-
-$(embedding_binary): $(wildcard embedding/*)
-	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT)/examples/embedding $(JULIA_BUILD_MODE) \
-                                JULIA="$(bindir)/julia$(JULIA_LIBSUFFIX)$(EXE)" BIN="$(outdir)" \
-                                SPAWN="$(spawn)" CC="$(CC)"
-
-clean:
-	-rm -f $(embedding_binary) $(embedding_binary)-debug
-
-.PHONY: all embedding clean
-
+release: # default target
+# forward all variables expected by the embedding example
+JULIA:=$(call spawn,$(JULIA_EXECUTABLE))
+BIN:=$(build_bindir)
```

Comment:
I didn't say this PR was perfect, I'm only working on making it much better than the situation that we decided to merge for now and then iterate on. If you're going to object to using `build_bindir`, I can also point out that this makefile shouldn't have ever been using the `libexecdir` variable since it isn't even a defined location. This was already discussed before it was merged (https://github.com/JuliaLang/julia/pull/21299#issuecomment-294204366).